ssh_check/tests/test_ssh.py

```diff
@@ -44,3 +49,44 @@ def test_ssh_bad_config(aggregator):
     # Check that we've closed all connections, if not we're leaking threads
     common.wait_for_threads()
     assert nb_threads == threading.active_count()
+
+
+@pytest.mark.parametrize(
+    'version, metadata',
+    [
+        (
+            'OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5',
+            {
+                'version.major': '7',
+                'version.minor': '7',
+                'version.release': 'p1',
+                'version.scheme': 'ssh_check',
+                'flavor': 'OpenSSH',
+            },
+        ),
+        (
+            'SSH-2.0-OpenSSH_8.1',
+            {'version.major': '8', 'version.minor': '1', 'version.scheme': 'ssh_check', 'flavor': 'OpenSSH'},
+        ),
+        (
+            'SSH-2.0-OpenSSH_7.4p1 Debian-10+deb9u2',
+            {
+                'version.major': '7',
+                'version.minor': '4',
+                'version.release': 'p1',
+                'version.scheme': 'ssh_check',
+                'flavor': 'OpenSSH',
+            },
+        ),
+    ],
+)
+def test_collect_metadata(version, metadata):
+    client = MagicMock()
+    client.get_transport = MagicMock(return_value=namedtuple('Transport', ['remote_version'])(version))
+
+    ssh = CheckSSH('ssh_check', {}, {}, list(common.INSTANCES.values()))
+    ssh.check_id = 'test:123'
+    with mock.patch('datadog_checks.base.stubs.datadog_agent.set_check_metadata') as m:
+        ssh._collect_metadata(client)
+        for key, value in metadata.items():
+            m.assert_any_call('test:123', key, value)
```

Comment:
Any reason not to use `datadog_agent.assert_metadata` ? Instead of 

```python
    with mock.patch('datadog_checks.base.stubs.datadog_agent.set_check_metadata') as m:
        ssh._collect_metadata(client)
        for key, value in metadata.items():
            m.assert_any_call('test:123', key, value)
```

Reply:
Thought it was only for integration tests